restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -1811,4 +1859,158 @@ private SipURI outboundInterface(String transport) {
         }
         return result;
     }
+
+    private boolean isFromIms(final SipServletRequest request) throws ServletParseException {
+        SipURI uri = (SipURI)request.getRequestURI();
+        return uri.getUser() == null;
+    }
+
+    private Registration findRegistration(javax.servlet.sip.URI regUri){
+        if(regUri == null){
+            return null;
+        }
+        String formattedNumber = null;
+        if (regUri.isSipURI()) {
+            formattedNumber = ((SipURI)regUri).getUser().replaceFirst("\\+","");
+        } else {
+            formattedNumber = ((TelURL)regUri).getPhoneNumber().replaceFirst("\\+","");
+        }
+        if(logger.isInfoEnabled()) {
+            logger.info("looking for registrations for number: " + formattedNumber);
+        }
+        final RegistrationsDao registrationsDao = storage.getRegistrationsDao();
+        List<Registration> registrations = registrationsDao.getRegistrations(formattedNumber);
+
+        if(registrations == null || registrations.size() ==0){
+            return null;
+        }
+        else{
+            return registrations.get(0);
+        }
+    }
+
+    private void imsProxyThroughMediaServer(final SipServletRequest request, final Client client,
+            final javax.servlet.sip.URI destUri, final String user, final String password, boolean isFromIms)
+                    throws IOException {
+        javax.servlet.sip.URI srcUri = request.getFrom().getURI();
+        if(logger.isInfoEnabled()) {
+            logger.info("imsProxyThroughMediaServer, isFromIms: " + isFromIms +
+                    ", destUri: " + destUri + ", srcUri: " + srcUri);
+        }
+        final Configuration runtime = configuration.subset("runtime-settings");
+        String destNumber = destUri.toString();
+        String rcml = "<Response><Dial>" + destNumber + "</Dial></Response>";
+
+        javax.servlet.sip.URI regUri = null;
+        if (isFromIms) {
+            regUri = destUri;
+        } else {
+            regUri = srcUri;
+        }
+
+        Registration reg = findRegistration(regUri);
+        if (reg == null) {
+            if(logger.isInfoEnabled()) {
+                logger.info("registrations not found");
+            }
+            final SipServletResponse response = request.createResponse(SC_NOT_FOUND);
+            response.send();
+            // We didn't find anyway to handle the call.
+            String errMsg = "Call cannot be processed because the registration: " + regUri.toString()
+                    + "cannot be found";
+            sendNotification(errMsg, 11005, "error", true);
+            return;
+        } else {
+            if (isFromIms) {
+                rcml = "<Response><Dial><Client>" + reg.getUserName() + "</Client></Dial></Response>";
+            }
+            boolean isFromWebRTC = false;
```

Comment:
@gvagenas 
you're right, this part of method is needless (it's a leftover from my work) - I remove it.